packages/docusaurus-theme-classic/src/theme/BlogPostItem/Header/Author/index.tsx

```diff
@@ -5,11 +5,70 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import type {SVGProps} from 'react';
 import React from 'react';
 import clsx from 'clsx';
 import Link, {type Props as LinkProps} from '@docusaurus/Link';
 
 import type {Props} from '@theme/BlogPostItem/Header/Author';
+import styles from './styles.module.css';
+
+type SocialProps = SVGProps<SVGSVGElement> & {
+  user: string;
+};
+
+function getUrl(user: string, baseUrl: string) {
+  return user.startsWith('https') ? user : `${baseUrl}${user}`;
+}
+
+function Twitter(props: SocialProps) {
+  const twitterUrl = getUrl(props.user, 'https://x.com/');
+  return (
+    // eslint-disable-next-line @docusaurus/no-html-links
+    <a
+      href={twitterUrl}
+      target="_blank"
+      rel="noreferrer noopener"
+      className={clsx(styles.socialIcon)}>
+      <svg
+        viewBox="0 0 256 209"
+        width="1.5em"
+        height="1.5em"
+        xmlns="http://www.w3.org/2000/svg"
+        className={clsx(styles.socialIcon)}
```

Comment:
remove CSS module from the reusable component

Parent can/should pass it as props